platform-sdk/docs/proposals/DS-0x/DS - Tss block signing - Cryptography APIs.md

```diff
@@ -0,0 +1,458 @@
+- **EC ([Elliptic-Curves](https://crypto.stanford.edu/pbc/notes/elliptic))**: "Elliptic" is not elliptic in the sense of an "oval circle". In the field "Fp", an EC is like a non-connected cloud of points. All points satisfy an equation and all operations are performed modulo p. Some elliptic curves are pairing-friendly.
+- **Bilinear Pairings**: Are mathematical functions used in cryptography to map two elements of different groups (in EC, the group is an elliptic curve) to a single value in another group in a way that preserves certain algebraic properties.
+- **Fields**: Mathematical structures where addition, subtraction, multiplication, and division are defined and behave as expected (excluding division by zero).
+- **Groups**: Sets equipped with an operation (like addition or multiplication) that satisfies certain conditions (closure, associativity, identity element, and inverses).
+- **Share**: Represents a piece of the public/private necessary elements to create or verify a signature. In TSS, a threshold number of shares is needed to validate a signature. To represent weight, any participant can have more than 1 share.
+- **Polynomial Commitment**: A polynomial commitment is a cryptographic commitment to a polynomial. It enables a committer to commit to a polynomial such that: a)The committer can later reveal the polynomial and prove that it was indeed the committed polynomial and b)The commitment allows verifying evaluations of the polynomial at specific points without revealing the entire polynomial.
+
+### Goals
+- **Usability:** Design a user-friendly library that is easy to integrate with consensus-node code and block-node code.
+- **EVM support:** Our implementation must be supported by EVM.
+- **Security:** Our APIS should use audited code and be able to pass security audits both internal and external.
+- **Flexibility**: Minimize the impact of introducing support for other elliptic curves. 
+- **Independent Release:** Design release cycle independent components that can be used both in the consensus node and in the future block node.
+
+### Non-Goals
+- Build our own cryptography library implementation that supports EC-curves from the gecko in Java.
+
+### External References
+- https://andrea.corbellini.name/2015/05/17/elliptic-curve-cryptography-a-gentle-introduction/
+- https://www.iacr.org/archive/asiacrypt2001/22480516.pdf
+- https://hackmd.io/@benjaminion/bls12-381#Motivation
+- https://www.johannes-bauer.com/compsci/ecc/
+
+## Changes
+
+### Core Behaviors
+
+The proposed TSS solution combines EC-Cryptography primitives that allow for short aggregable signatures, assignation of multiple shares per shareholder to represent participants' weight,
+and combines Shamir's secret sharing with ElGamal public key encryption and groth21 for zero-knowledge proofs.
+
+#### Description of the Scheme
+
+An operator will generate a long-living persisting ec private/public key pair for each participant of the scheme.
```

Comment:
`long-living persisting` is less clear than simply `persistent`

this phrase also appears elsewhere in this doc, if you decide to make the change